### PR TITLE
Feature/authentication

### PR DESCRIPTION
**Pull request by protodaniel**

### CODE REVIEW

File: .awsconfig/awsconfig.json [hunk 1]
```diff
@@ -0,0 +1,25 @@
+{
+  "Auth": {
+    "region": "ap-southeast-2",
+    "userPoolId": "ap-southeast-2_Je8iCzK3z",
+    "userPoolWebClientId": "579fl31q38092s0if1i6dutnpg",    
+    "authenticationFlowType": "USER_PASSWORD_AUTH",
+    "redirectSignIn": "http://localhost:3000",
+    "redirectSignOut": "http://localhost:3000",
```
Comment (willdawsonme) [inline .awsconfig/awsconfig.json:8]: Is there a way to have this come from config/environment variables? These redirects won't work in other environments right?

File: package.json [hunk 2]
```diff
@@ -12,13 +12,17 @@
     "build-storybook": "build-storybook -s ./public"
   },
   "dependencies": {
+    "@aws-amplify/ui-components": "^1.7.8",
```
[the remark this answers is not in the document]
Comment (protodaniel) [reply package.json:15]: yup, removing both of the unused ones

File: src/hooks/api/useDeleteEventMutation.ts [hunk 3]
```diff
@@ -13,18 +12,10 @@ interface UseDeleteEventMutationVariables {
...
-      `/events/${id}`,
-      {
-        token,
-      },
-      {
-        method: 'DELETE',
-      },
-    );
+    await apiFetch(`/events/${id}`, {
+      method: 'DELETE',
```
[the remark this answers is not in the document]
Comment (protodaniel) [reply src/hooks/api/useDeleteEventMutation.ts:17]: Huh, I don't remember doing deleteMutation... will add an error message and fix it

File: src/types/index.ts [hunk 4]
```diff
@@ -151,3 +152,30 @@ declare module '@mui/material/IconButton' {
...
+  vimeo?: {
+    rtmpKey: string;
+  };
+}
+
+/**
+ * API
+ */
+
+export interface ApiResponse {
```
[the remark this answers is not in the document]
Comment (protodaniel) [reply src/types/index.ts]: Was wrapping it erroneously in some useMutation. It's gone now.

File: src/routes/Home/Home.tsx [hunk 5]
```diff
@@ -1,26 +1,25 @@
...
   title: '',
   rtmpKey: '',
 };
 
 export default function Index(): ReactElement {
-  const accessToken =
-    new URLSearchParams(useLocation().search).get('accessToken') || '';
-  GoLiveAPI.setAuthToken(accessToken);
+  const createUserMutation = useCreateEventMutation();
+  const { user } = useAuthState();
```
[the remark this answers is not in the document]
Comment (protodaniel) [reply src/routes/Home/Home.tsx:22]: yup!

File: src/routes/Manage/Event/Event.tsx [hunk 6]
```diff
@@ -5,12 +5,14 @@ import React, { useState } from 'react';
...
 import Page from 'src/components/Page';
 import useEventQuery from 'src/hooks/api/useEventQuery';
+import useAuthState from 'src/hooks/useAuthState';
 import PageTab, { PAGE_TAB } from './PageTab';
 import PublishButton from './PublishButton';
 import StreamTab, { STREAM_TAB } from './StreamTab';
 
 const Event = (): React.ReactElement => {
   const { id } = useParams();
+  const user = useAuthState();
```
[the remark this answers is not in the document]
Comment (protodaniel) [reply src/routes/Manage/Event/Event.tsx:15]: Si

File: src/components/AuthProvider/AuthProvider.tsx [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+const AuthProvider: React.FC = ({ children }) => {
+  const initialAuthState: AuthState = {
+    status: 'idle',
+    user: undefined,
+  };
+
+  type ActionType = {
+    type: string;
+    payload?: any;
+  };
```
[the remark this answers is not in the document]
Comment (protodaniel) [reply src/components/AuthProvider/AuthProvider.tsx:18]: deal

File: src/types/index.ts [hunk 8]
```diff
@@ -151,3 +152,54 @@ declare module '@mui/material/IconButton' {
...
+    };
+  };
+  blocks: Array<unknown>;
+}
+
+/**
+ * API
+ */
+
+export interface ApiResponse<T> {
```
[the remark this answers is not in the document]
Comment (protodaniel) [reply src/types/index.ts:199]: eh? Double comment!

File: src/types/index.ts [hunk 9]
```diff
@@ -151,3 +152,54 @@ declare module '@mui/material/IconButton' {
...
+  customerId: string;
+  streamTitle: string;
+  startTime: number;
+  type: string; // todo: make this a type that only takes `single | recurring`
+  vimeo?: {
+    rtmpKey: string;
+  };
+}
+
+export interface CreateEventApiResponse {
```
[the remark this answers is not in the document]
Comment (protodaniel) [reply src/types/index.ts:171]: Ah yeah nice! Didn't see the Event one. Thanks!

File: .awsconfig/awsconfig.json (hunk 1 above)
Comment (protodaniel) [reply .awsconfig/awsconfig.json:8]: yeah, time to fix config. I'll bring in a bunch of the setup stuff

**Pull request merged**